okteto/okteto
# [Fix] [Pull-First-Approach] Search first in dev registry

# Proposed changes

Fixes https://github.com/okteto/app/issues/6710

## How to reproduce the bug

1. Use context: `demo.okteto.dev`
2. Clone https://github.com/okteto/movies-with-helm
3. Build the CLI from `master` or use the latest release (`2.16.3`), and run `okteto build` in `movies-with-helm`
4. Notice how the image is being built successfully (it's probably already built, and it's okay)
5. Now make a change (eg `echo hi > hi.txt`)
6. Run again `okteto build` and notice the cmd succeeding again
7. Now run `okteto deploy`, wait until complete, then go to the UI
8. Inspect the YAML of the movies API deployment, search for the `image:` and notice that it's using an image from the `okteto` namespace instead of your own, example `image: registry.demo.okteto.dev/okteto/movies-with-helm....`).

The `image:` should have been pointing to your own namespace, because by making the repo dirty, and building a new image, you now have a tag for that change but it wasn't used at deploy time.

## How to test the fix

1. Using this branch, build the CLI
2. Using the same repository `movies-with-helm` and keeping the change you made run the new version of the CLI
3. Check again the `image:` property in the Dashboard and notice how it now shows: `image: registry.demo.okteto.dev/<your namespace>/movies-with-helm-api...`
